rrs/tools/rrs_upgrade_history.py

```diff
@@ -0,0 +1,293 @@
+
+                if options.commit:
+                    initial = False
+                    since = options.commit
+                    since_option = ['%s^..%s' % (options.commit, options.commit)]
+                elif maintplanbranch.upgrade_rev and not options.fullreload:
+                    initial = False
+                    since = maintplanbranch.upgrade_date
+                    since_option = ['%s..origin/master' % maintplanbranch.upgrade_rev]
+                else:
+                    initial = True
+                    since = options.since
+                    since_option = ['--since=%s' % since, 'origin/master']
+
+                repo = git.Repo(repodir)
+                if repo.bare:
+                    logger.error('Repository %s is bare, not supported' % repodir)
+                    continue
+
+                commits = utils.runcmd(['git', 'log'] + since_option + ['--format=%H %ct', '--reverse'],
+                                    repodir,
+                                    logger=logger)
+                commit_list = commits.split('\n')
+
+                bitbake_map = {}
+                def remap_range(start, end, replacewith=None):
+                    if replacewith is None:
+                        replacewith = end
+                    bitbake_commits = utils.runcmd(['git', 'rev-list', '%s^..%s^' % (start, end)],
+                                        bitbakepath,
+                                        logger=logger)
+                    bitbake_commit_list = bitbake_commits.splitlines()
+                    for commit in bitbake_commit_list:
+                        bitbake_map[commit] = replacewith
+
+                # Filter out some bad commits
+                remap_range('fef18b445c0cb6b266cd939b9c78d7cbce38663f', '39780b1ccbd76579db0fc6fb9369c848a3bafa9d')
+                remap_range('5796ed550d127853808f38257f8dcc8c1cf59342', '547128731e62b36d2271c4390b3fee2b16c535dc')
+
+                if options.stop_commit and (options.stop_commit not in [x.split()[0] for x in commit_list]):
+                    logger.error('Stop commit %s is not in repository %s' % (options.stop_commit, repodir))
+                    sys.exit(1)
+
+                if initial:
+                    logger.debug("Adding initial upgrade history ....")
+
+                    ct, ctepoch = commit_list.pop(0).split()
+                    ctdate = datetime.fromtimestamp(int(ctepoch))
+                    run_internal(maintplanbranch, ct, ctdate, options, logger, bitbake_map, initial=True)
+
+                if layerbranch.vcs_subdir:
+                    layersubdir_start = layerbranch.vcs_subdir
+                    if not layersubdir_start.endswith('/'):
+                        layersubdir_start += '/'
+                else:
+                    layersubdir_start = ''
+                logger.debug("Adding upgrade history from %s to %s ..." % (since, datetime.today().strftime("%Y-%m-%d")))
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Naive datetime objects are treated by many datetime methods as local times, We recommend that you create aware timestamps with the `now` or `fromtimestamp` functions using the `tzinfo` to attribute to represent times in UTC. Do not use the [datetime.utcnow()](https://docs.python.org/3/library/datetime.html#datetime.datetime.utcnow) because it returns only a `naive` datetime object. 

[Learn more](https://docs.python.org/3/library/datetime.html#aware-and-naive-objects)